NitroxModel/Packets/LiveMixinHealthChanged.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.Util;
+using ProtoBufNet;
+
+namespace NitroxModel.Packets
+{
+    [Serializable]
+    [ProtoContract]
+    public class LiveMixinHealthChanged : Packet
+    {
+        [Serializable]
+        class DamageTakenData
+        {
+            public NitroxVector3 Position { get; set; }
+
+            public ushort Damagetype { get; set; }
+
+            public Optional<NitroxId> DealerId { get; set; }
+        }
+        Optional<DamageTakenData> damageTakenData;
+        public NitroxTechType TechType { get; }
+
+        public NitroxId Id { get; set; }
+
+        public float LifeChanged { get; set; }
+
+        
+        public NitroxVector3 Position { get { return damageTakenData.HasValue ? damageTakenData.Value.Position : default; } }
+
+        public ushort Damagetype { get { return damageTakenData.HasValue ? damageTakenData.Value.Damagetype : default; } }
+
+        public Optional<NitroxId> DealerId { get { return damageTakenData.HasValue ? damageTakenData.Value.DealerId : default; } }
```

Comment:
minor: Is this model also intended to be used on its own?  Can it just be fields here?

Reply:
Had them just as fields before but NitroxVector3 and ushort cannot be used with Optional. But we don't need those fields for Health healed. Therefore, i separated them into an internal class that can be set empty to (hopefully) reduce package size.